src/lib.rs

```diff
@@ -53,11 +55,21 @@ fn format_file(config: &MdsfConfig, input: &str) -> (bool, String) {
                 if is_snippet {
                     print_line_info(language, line_index + 1, line_index + snippet_lines + 1);
 
+                    if language == Language::Go && !code_snippet.contains("package ") {
```

Comment:
This could be a source of bug

````markdown
```go

// my package is rich
var err string
```
````

I would say the good way to test is the fact there is one line starting with "\s*package "

Please note, it's not necessarily the first as you can get comments in front of the package declaration, and the space in frot because copy pasted code could be misaligned with tab or spaces

````markdown
```go
    // foo provides teapot
    // yeah
    package foo
  
            // foo bad idented
    type foo    string
  type bar               int
```
```

Reply:
That's why I talked about detecting

^\s*package\s+\w

But it will catch this

````markdown
```go

/* foo

    package foo

*/

type   foo    string

```
````

I think you should add it to an false negative edge case in the test. As a reminder for a later refactoring.

Something like: todo compromise

Here you are getting feedback only from me on this feature

Reply:
It is definitely helping a lot 😃 
Not a go user so didn't really notice until you mentioned it.

Reply:
I'm not a Rust developer 😅

But I'm very good at reverse engineering code 😜

I'm helping you because it will help you to build a great too I could use to format tons of documentation I have.

Reply:
I added the regex stuff in #159.   Gonna skip the multiline comment stuff for now. I am hoping it is an edge case that people wont hit 😅  Most likely gonna remove the regex stuff altogether and just check the snippet directly.

Reply:
I also added an issue so I don't forget :D  (#160)